src/middlewared/middlewared/etc_files/rancher/k3s/flags.py

```diff
@@ -50,3 +51,10 @@ def render(service, middleware):
             'protect-kernel-defaults': True,
             'disable': [] if config['servicelb'] else ['servicelb'],
         }))
+
+    # TODO: See what we have to do make proxy work for pulling images
```

Comment:
Seems to be doable with an env-var:

https://github.com/rancher/rke2/issues/517